src/app/digital/utils/ExpressionParser/GenerateInputTree.ts

```diff
@@ -124,15 +159,25 @@ export function GenerateInputTree(tokens: Token[], ops: Record<TokenType, string
     const index = ret.index;
     if (index < tokens.length) {
         if (tokens[index].type === ")")
-            throw new Error("Encountered Unmatched \"" + ops[")"] + "\"");
+            throw new Error(`Encountered Unmatched "${ops[")"]}"`);
 
         const prev = tokens.slice(0, index) // Decrementing through the array starting at right before the returned index
                            .reverse()
                            .find(token => token.type === "input") as InputToken;
         const next = tokens.slice(index)
                            .find(token => token.type === "input") as InputToken;
-        if (prev && prev.name && next && next.name)
-            throw new Error("No valid operator between \"" + prev.name + "\" and \"" + next.name + "\"");
+        if (prev && prev.name && next && next.name) {
+            let errorMessage = `No valid operator between "${prev.name}" and "${next.name}"`;
+            for (const format of Formats) {
+                Object.entries(format.ops).forEach(([tokenType, op]) => {
+                    if (op === prev.name)
+                        errorMessage += `\nDid you mean to use "${ops[tokenType as TokenType]}" instead of "${prev.name}"?`
+                    if (op === next.name)
+                        errorMessage += `\nDid you mean to use "${ops[tokenType as TokenType]}" instead of "${next.name}"?`
+                });
+            }
+            throw new Error(errorMessage);
```

Comment:
This function is starting to get quite long, hopefully the previous suggestions will lead to a smaller size, but also here you could probably make a `BuildErrorMessage` function or something and keep the `throw new Error` as a 1-liner